### ISSUE TITLE
Deserializing big JSON genesis state is very slow

### ISSUE THREAD

**Issue by crystalin** [paritytech/substrate#11867]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

We are often using exported state to run "forked" network for testing. We noticed that it takes a long time to load the genesis of those network. The file is big (~5GB) but the loading time is over 1 hour for those, which I think is due to a bug (serde_json specifies they could do deserialization at up to 1GB/s).

I tried to investigate and here are the few things I noticed:
* The JSON file seemed to be half read almost instantly (monitoring reports 2.7 GB disk read from the first 5 seconds) and then only few disk reads every X seconds.
* The process doesn't increase its memory usage much (around 150MB after 1 minute, and slowly increasing ~30MB/minute)  
* The CPU is at 100% (single-threaded it seems, no matter the OS/machine being used)
* All the CPU is taken into the system read :/ 

<img width="1299" alt="image" src="https://user-images.githubusercontent.com/329248/179866003-7882c7e9-26ab-42aa-8820-840160ecb4e5.png">

### Steps to reproduce

(You can easily reproduce it by doing an export-state of polkadot or kusama and using it as a --chain argument)

Here is a sample file you can use to test (using moonbeam):
```
wget https://s3.us-east-2.amazonaws.com/snapshots.moonbeam.network/moonriver/latest/moonriver-state.json
```

Here is how to test it:
```
wget https://github.com/PureStake/moonbeam/releases/download/v0.25.0/moonbeam
chmod uog+x moonbeam

./moonbeam --tmp --chain moonriver-state.json
# Wait for 1 hour or more
```

**Comment by crystalin** [comment]
It seems someone already worked on it, but was reverted maybe?
https://github.com/paritytech/substrate/pull/10137/files

**Comment by nazar-pc** [comment]
Did you check Git history to see why it was reverted?

**Comment by crystalin** [comment]
@nazar-pc it wasn't revert, it is just that the file is read in 2 places:
https://github.com/paritytech/substrate/blob/master/client/chain-spec/src/chain_spec.rs#L61 ( fixed in the PR I just proposed #11868)
and 
https://github.com/paritytech/substrate/blob/master/client/chain-spec/src/chain_spec.rs#L310 (which is the fix I referred previously)
